### PR TITLE
Matek f405 wing and OSD support

### PR DESCRIPTION
**Pull request by hendrixgr**
I think this time everything should be clean and suitable for merging.
Gautier thank you for being patient with me.
Chris

### CODE REVIEW

**Review by gautierhattenberger — CHANGES_REQUESTED** [review]
great job, I only have a few minor comments, and if you don't have time, it is already good to merge

File: conf/boards/matek_f405_wing_v1.makefile [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Hey Emacs, this is a -*- makefile -*-
+#
+# omnibus_f4_aio.makefile
+#
+# https://www.openpilot.org/products/openpilot-revolution-platform/
```
[the remark this answers is not in the document]
Comment (hendrixgr) [reply conf/boards/matek_f405_wing_v1.makefile:5]: No, i will fix it.

File: conf/airframes/matek_f405_wing/Easyglider_matek_f405_wing.xml [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+      <module name="ahrs" type="float_dcm"/>
+      <module name="ins" type="alt_float">
+        <configure name="USE_MAGNETOMETER" value="0"/>
+        <define name="AHRS_USE_GPS_HEADING" value="1"/>
+        <define name="AHRS_GRAVITY_UPDATE_COORDINATED_TURN"/>
+      </module>
+      <module name="spi" type="master"/>
+      <module name="nav_line.xml"/>
+      <module name="gps_ubx_ucenter.xml"/> 
+      <module name="baro_bmp280_i2c.xml">
```
[the remark this answers is not in the document]
Comment (hendrixgr) [reply conf/airframes/matek_f405_wing/Easyglider_matek_f405_wing.xml:57]: I am reading about it right now, is it ok if don't want to use the auto load method?

[the remark this answers is not in the document]
Comment (hendrixgr) [reply conf/airframes/matek_f405_wing/Easyglider_matek_f405_wing.xml]: I am trying to make it work but i get anerror
Paparazzi version v5.17_devel-83-g972b17a-dirty
-----------------------------------------------------------------------
#######################################
# BUILD AIRCRAFT=Easyglider_Matek, TARGET ap
#######################################
Aircraft generator: 'Easyglider_Matek' for target 'ap'
Parsing airframe '/home/hendrix/paparazzi/conf/airframes/CEF/Easyglider_matek_f405_full.xml', sorting by targetFatal error: exception Module.Module_not_found("board_matek_wing.xml")
Makefile.ac:201: recipe for target 'ap.ac_h' failed
make: *** [ap.ac_h] Error 2
make: Leaving directory '/home/hendrix/paparazzi'
FAILED 'make -C /home/hendrix/paparazzi -f Makefile.ac AIRCRAFT=Easyglider_Matek ap.compile' with code 2

[the remark this answers is not in the document]
Comment (hendrixgr) [reply conf/airframes/matek_f405_wing/Easyglider_matek_f405_wing.xml]: Never mind i was typing the file name wrong....

[the remark this answers is not in the document]
Comment (gautierhattenberger) [reply conf/airframes/matek_f405_wing/Easyglider_matek_f405_wing.xml]: you don't have to make this meta-module, it is just a possibility to make a cleaner airframe file when using a default config.
if you are fine like this, I can merge

[the remark this answers is not in the document]
Comment (hendrixgr) [reply conf/airframes/matek_f405_wing/Easyglider_matek_f405_wing.xml]: Meta module done but i think i overdid it....

[the remark this answers is not in the document]
Comment (hendrixgr) [reply conf/airframes/matek_f405_wing/Easyglider_matek_f405_wing.xml]: I think i finished....

[the remark this answers is not in the document]
Comment (gautierhattenberger) [reply conf/airframes/matek_f405_wing/Easyglider_matek_f405_wing.xml:57]: maybe that was a bit to much indeed :) 
I suggest to keep it limited to internal sensors, so it is easier to reuse for other people, with different gps, ahrs, et so on (you can still have your personal meta-module if it makes your configuration easier, thought).
You can find my proposal in the `matek` branch of my repo: https://github.com/gautierhattenberger/paparazzi/commit/647ec61a9a16966b523bbf5f535ce85679785c9a
If you agree to these changes, I can add that to your PR and merge.

**Comment by noether** [comment]
@hendrixgr 

Thanks for this PR, the Matek looks interesting, and I think I will give it a try :P.

File: conf/airframes/matek_f405_wing/Easyglider_matek_f405_wing.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (hendrixgr) [reply conf/airframes/matek_f405_wing/Easyglider_matek_f405_wing.xml:57]: Of course i agree please change whatever you want.
Chris

**Comment by gautierhattenberger** [comment]
merged with 8b56049d2145e27e191a0eb9213b3f27406ca773

File: conf/airframes/matek_f405_wing/Easyglider_matek_f405_wing.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (gautierhattenberger) [reply conf/airframes/matek_f405_wing/Easyglider_matek_f405_wing.xml:57]: I have just moved some modules back to the airframe files, but it should work like before. I merged it directly to master. Thanks again for your work, these Matek boards are interesting.

**Pull request closed without merging**